### PR TITLE
r2retdec

### PR DESCRIPTION
**Pull request by securisec**
local retdec open source bridge for r2

### CODE REVIEW

**Comment by radare** [comment]
Why not name the package retdec-local?

File: db/r2decompile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Optional arguments:
+  -h, --help  Show this help message and exit.
+  -t TMP      Set temp file for decompiled code
+  -p          Print dicompilation to stdout
+
+Invoke from inside r2 shell with $dec
+"
+
+R2PM_INSTALL() {
+    cd ${R2PM_GITDIR}r2decompile
```
Comment (radare) [inline db/r2decompile]: Check for errors with || ..

File: db/r2decompile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    npm install
```
Comment (radare) [inline db/r2decompile]: Check errors here

File: db/r2decompile (hunk 1 above)
Comment (radare) [reply db/r2decompile]: Add a slash

File: db/r2decompile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    cd -
+	echo '$'dec="#"'!'"pipe node ${R2PM_GITDIR}r2decompile/r2decompile.js" >> ~/.radare2rc || exit 1
+	echo "\n[+] r2decompile has been installed"
```
Comment (radare) [inline db/r2decompile]: Dont use escape codes with echo. Its not portable. Remove the \n

**Comment by securisec** [comment]
if you insist, i can change it. Too many changes in everything if i rename it

File: db/r2decompile [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  -p          Print dicompilation to stdout
+
+Invoke from inside r2 shell with $dec
+"
+
+R2PM_INSTALL() {
+    cd ${R2PM_GITDIR}r2decompile
+    npm install
+    cd -
+	echo '$'dec="#"'!'"pipe node ${R2PM_GITDIR}r2decompile/r2decompile.js" >> ~/.radare2rc || exit 1
```
Comment (radare) [inline db/r2decompile]: Use bindir from r2pm to honor -g and homedir properly

Comment (radare) [reply db/r2decompile]: Ok thats just an echo. You can put that script in bindir from r2pm and make jt accessible from r2 without having to set the full path

File: db/r2decompile (hunk 1 above)
Comment (securisec) [reply db/r2decompile]: done

File: db/r2decompile (hunk 2 above)
Comment (securisec) [reply db/r2decompile]: done

File: db/r2decompile (hunk 3 above)
Comment (securisec) [reply db/r2decompile]: done

File: db/r2decompile (hunk 4 above)
Comment (securisec) [reply db/r2decompile]: not sure what you mean. all my other plugins follow this same format in r2pm

**Comment by radare** [comment]
Yes i think so. Maybe in your repo can name it after r2 to make that clear that is a connection between r2 and local retdec installation.

Also will be good to have a package to install retdec

**Comment by radare** [comment]
ln -fs ${R2PM_GITDIR}r2decompile/r2decompile.js${R2PM_GITDIR}/retdec-local

And then call it like this:

#!pipe retdec-local

No need to specify node or full path. Its way shorter and more generic

**Comment by securisec** [comment]
OK, changed the package name to r2retdec. I am not a fan of including local in there. All other changes made besides the call using an alias. I find the symlink method is not super consistant, and prefer the alias method. 

As for package to install retdec, this is not suitable at the moment. Avast does not provide a release for linux, and building and compiling retdec can be quite a long process. I am considering including a Vagrant file that automates some of this process, but due to the amount of memory that the compiling process takes, that is not very suitable either. Hence the decision to go with a `.` config file for the decompiler path

**Comment by securisec** [comment]
added to use `pdc` to populate data if the event of 64 bit, and function summary shortcuts like strings, xrefs and calls. decomplication can also now be printed to stdout using an option.

**Comment by radare** [comment]
the package is still named r2decompile, please, rename it to r2retdec-local or whatever

**Comment by securisec** [comment]
I renamed everything to `r2retdec`. The I edited all the changes to the pull request and on the repo to reflect the new `r2retdec` name https://github.com/securisec/r2retdec

**Comment by radare** [comment]
nope. see your PR `db/r2decompile`

**Comment by securisec** [comment]
fixed that. missed that one

**Comment by radare** [comment]
thanks!

**Comment by securisec** [comment]
thanks!

**Pull request merged**